client/src/pages/userPage.tsx

```diff
@@ -73,6 +80,32 @@ const UserPage: FC = () => {
     setIsEditMode(false);
   };
 
+  const handleDelete = () => {
+    const toastDeleteUserId = toast.loading('Loading...');
+    deleteUser({ variables: { name: paramsUserName } })
+      .then((res) => {
+        toast.success('Successfully deleted', {
+          id: toastDeleteUserId,
+        });
+        cache.modify({
+          fields: {
+            users: (existingUserRefs: User[], { readField }) =>
+              existingUserRefs.filter(
+                (userRef: User) =>
+                  res.data?.deleteUser.user.id !== readField('id', userRef),
```

Comment:
サーバーから削除したuserの情報を返さなくても，
`paramsUserName`が存在するのだから，これを使用すれば良いのでは？
```suggestion
                  paramsUserName !== readField('name', userRef),
```